python/desktop_client/DesktopClient.py

```diff
@@ -0,0 +1,315 @@
+
+    def call_server_method(self, name, data=None):
+        """
+        Make a call to a WebSocket server method and return the reply as a python dict.
+
+        Arguments:
+            name {str} -- Name of the server method
+
+        Keyword Arguments:
+            data {dict} -- Arguments of the server method (default: {None})
+
+        Returns:
+            dict -- Reply from the server as a python object (reply json is part).
+        """
+        # Get the server method as a Dict
+        resp = json.loads(self._call_server_method(name, data))
+        return resp.get("reply", "")
+
+    @property
+    def _desktop_connection(self):
+        """
+        Returns the currently active websocket connection to the Shotgun WebSocket server.
+
+        If there's no active connectionn, the function builds a new connection and do the
+        handshake so it's ready to use.
+
+        Returns:
+            WebSocket -- Active websocket connection to the Shotgun WebSocket server.
+        """
+        try:
+            if self._connection:
+                # Ping the server to make sure the connection is still valid.
+                try:
+                    self._connection.ping()
+                except:
+                    logger.debug(
+                        "Failed to reuse the active connection. Destroying the connection.")
+                    self._connection = None
+
+            if not self._connection:
+                os.environ["WEBSOCKET_CLIENT_CA_BUNDLE"] = self.ca_file.name
+                self._connection = websocket.create_connection(
+                    "wss://shotgunlocalhost.com:{}".format(
+                        self.shotgun_create_websocket_port)
+                )
+
+                self._do_websocketserver_handshake()
+        except:
+            logger.debug(
+                "Failed to get a valid websocket connection")
+            self._connection = None
+            raise
+
+        return self._connection
+
+    @property
+    def _current_user(self):
+        """
+        Get the currently authenticated user.
+
+        Returns:
+            ShotgunUser -- Currently authenticated user.
+        """
+        return sgtk.get_authenticated_user()
+
+    @property
+    def _shotgun_connection(self):
+        """
+        Get a new Shotgun api connection using the currently authenticated user.
+
+        Returns:
+            Shotgun -- Shotgun connection.
+        """
+        return self._current_user.create_sg_connection()
+
+    def _send(self, payload):
+        """
+        Send a payload to the server.
+
+        The payload is encrypted using the WebSocket server secret, if available.
+
+        Arguments:
+            payload {str} -- Payload to send to the server.
+        """
+        try:
+            p = payload
+
+            if self._secret:
+                p = self._secret.encrypt(p)
+
+            self._desktop_connection.send(p)
+        except:
+            pass
```

Comment:
Could you log an error here saying what we couldn't send and the exception message so we can support this more easily?